fmralign/hyperalignment/local_template.py

```diff
@@ -0,0 +1,262 @@
+                if adjust_ns:
+                    XX /= np.sqrt(ns)
+                return XX.astype(np.float32), cc
+            else:
+                U, s, Vt = randomized_svd(
+                    X,
+                    (n_components if n_components is not None else min(X.shape)),
+                    random_state=0,
+                )
+                if adjust_ns:
+                    XX = U[:, :n_components] * (
+                        s[np.newaxis, :n_components] / np.sqrt(ns)
+                    )
+                else:
+                    XX = U[:, :n_components] * (s[np.newaxis, :n_components])
+                cc = Vt[:n_components].reshape(-1, ns, nv)
+                return XX.astype(np.float32), cc
+        except:  # noqa: E722
+            return PCA_decomposition(
+                X,
+                n_components=n_components,
+                flavor="svd",
+                adjust_ns=adjust_ns,
+                demean=demean,
+            )
+    elif flavor == "svd":
+        U, s, Vt = safe_svd(X)
+        if adjust_ns:
+            XX = U[:, :n_components] * (s[np.newaxis, :n_components] / np.sqrt(ns))
+        else:
+            XX = U[:, :n_components] * (s[np.newaxis, :n_components])
+        cc = Vt[:n_components].reshape(-1, ns, nv)
+        return XX.astype(np.float32), cc
+    else:
+        raise NotImplementedError
+
+
+def compute_PCA_template(X, sl=None, n_components=None, flavor="sklearn", demean=False):
+    """
+    Compute the PCA template from the input data.
+
+    Parameters:
+    -----------
+
+    - X (ndarray):
+        The input data array of shape (n_samples, n_features, n_timepoints).
+    - sl (slice, optional):
+        The slice of timepoints to consider. Defaults to None.
+    - n_components (int, optional):
+        The maximum number of principal components to keep. Defaults to None.
+    - flavor (str, optional):
+        The flavor of PCA algorithm to use. Defaults to "sklearn".
+    - demean (bool, optional):
+        Whether to demean the data before performing PCA. Defaults to False.
+
+    Returns:
+    --------
+
+    XX (ndarray):
+        The PCA template array of shape (n_samples, n_features, n_components).
+    """
+    if sl is not None:
+        X_ = X[:, :, sl]
+    else:
+        X_ = X
+    n = min(X_.shape[1], X_.shape[2])
+    n_components = min(n, n_components)
+    XX, cc = PCA_decomposition(
+        X_, n_components=n_components, flavor=flavor, adjust_ns=True, demean=demean
+    )
+    return XX.astype(np.float32)
+
+
+def compute_PCA_var1_template(
+    dss, sl=None, n_components=None, flavor="sklearn", demean=True
+):
+    if sl is not None:
+        dss = dss[:, :, sl]
+    XX, cc = PCA_decomposition(
+        dss, n_components=n_components, flavor=flavor, adjust_ns=False, demean=demean
+    )
+    w = np.sqrt(np.sum(cc**2, axis=2)).mean(axis=1)
+    XX *= w[np.newaxis]
+    return XX
+
+
+def compute_procrustes_template(
```

Comment:
I'd naively think this would go under template alignment. Indeed, `TemplateAlignment` can handle `scaled_orthogonal` alignments. Is there a reason to add this additional function here ?

Reply:
Comes from feilong's code but you are right. In fact there is no need for procrustes alignment in the INT model, so I will delete this code